doc/releases/changelog-dev.md

```diff
@@ -238,6 +240,9 @@
   >>> H =  qml.ops.dot(coeffs, ops)
   ```
   
+* Add `typing.Tensor` type.
```

Comment:
I'm hesitant to call this `Tensor` given `qml.operation.Tensor`.

Reply:
I changed it to `TensorLike`! Forgot to change the changelog